### ISSUE TITLE
January Update broke UI (Vulkan), so slow that matchmaking is impossible

### ISSUE THREAD

**Issue by alleey** [ValveSoftware/Dota-2#1881]
#### Your system information

* System information from steam (`Steam` -> `Help` -> `System Information`) in a [gist](https://gist.github.com/): https://gist.github.com/alleey/24d1d966f2c265b3c023b84e109f50a6
* Have you checked for system updates?: Yes
* Are you using the latest stable video driver available for your system? Yes
* Have you verified the game files?: Yes

#### Please describe your issue in as much detail as possible:
Since after the most recent update, which added DPC Live, DOTA UI has become extremely slow to transition from one screen to another. The initial load takes 2/3 minutes some times. Loading a hero model takes several seconds and the screen freezes several times during that. Party matches are IMPOSSIBLE, even bot matches, because the"Players joining ..." always fails due to the UI freezing. It takes another 30 secs of blank screen to take you back to main menu with the BACK TO LOBBY state.

The ONLY thing that works is Solo bot match and the UI is so slow that the whole strategy phase (30 secs) plays like a stuttering man's talk, only a few items can be selected during that time. 

Gameplay is OK, its only the Menus/UI that is a showstopper. Even Last Hit Trainer UI runs into the same stuns.

#### Steps for reproducing this issue:

1. Launch dota 2, add someone to party
2. Create a bot match
3. Player failed to connect message blinks (only the Linux player)
4. Screen goes blank after countdown to -10 (yes negative countdown 0 -> -10 on the Linux PC)
5. After some 30/60 secs main menu appears with BACK TO LOBBY button

**Comment by alleey** [comment]
I tried with OpenGL and that worked fine. Seems like Vulkan Shaders take so much time to compile that things timeout.

**Comment by Turmfalke2** [comment]
Your system information are incomplete. You copied it from here?

![screen02](https://user-images.githubusercontent.com/5174562/106354234-7b957e80-62f0-11eb-816c-1da19237631d.png)

**Comment by alleey** [comment]
@Turmfalke2 I did, and I just removed the extra info and added/fixed the url etc. It's an exact macth now

![image](https://user-images.githubusercontent.com/2000085/106622180-262dcb80-6595-11eb-8f1c-5413d7d3005f.png)

**Comment by Turmfalke2** [comment]
No, you pressed on 'About Steam'. Please press on 'System Information'. It should open a window similar to this.

![screen02](https://user-images.githubusercontent.com/5174562/106781732-a9105e00-6649-11eb-9386-a47da5aca787.png)

**Comment by alleey** [comment]
My bad. Updated. I hope I got it right this time.

**Comment by cakesale** [comment]
I'd personally start by upgrading Ubuntu to latest version (20.10) and try and get onto the latest version of mesa, will probably have to add a PPA for this.

I also see your system doesn't seem to have any graphics card besides the internal card? Is that correct? If so, turn down all the graphics settings and lower the resolution.

**Comment by shoaib-xo** [comment]
Already using lowest Graphic settings. With OpenGL the standard Bots and Online games work, but the Custom games still hang at around the same spot. This is where it freezes and how it looks like. Sometimes the hero model appears, sometimes the match starts and it freezes immediately. 
For some game types where you can restart your computer and rejoin, everything works fine.

![Screenshot_2021-03-05_21-15-55](https://user-images.githubusercontent.com/37289032/110142897-75ffec80-7df8-11eb-843d-8ef3c29a994b.png)

The only way out is to press the power button and use keyboard to relogin. Login again, goto task manager and kill the following.

![image](https://user-images.githubusercontent.com/37289032/110143641-40a7ce80-7df9-11eb-841f-fda4ff3d3b25.png)

With Vulkan even online match making is not possible. There is no Freeze, its terribly slow. 

OpenGl or Vulkan, the problem is always around the hero selection event.

**Comment by alleey** [comment]
Already using lowest Graphic settings. With OpenGL the standard Bots and Online games work, but the Custom games still hang at around the same spot. 

This is where it freezes and how it looks like. Sometimes the hero model appears, sometimes the match starts and it freezes immediately. 

![Screenshot_2021-03-05_21-15-55](https://user-images.githubusercontent.com/37289032/110142897-75ffec80-7df8-11eb-843d-8ef3c29a994b.png)

The only way out is to press the power button and use keyboard to relogin. Login again, goto task manager and kill the following.

![image](https://user-images.githubusercontent.com/37289032/110143641-40a7ce80-7df9-11eb-841f-fda4ff3d3b25.png)

With Vulkan even online match making is not possible. There is no Freeze, its terribly slow. 

OpenGl or Vulkan, the problem is always around the hero selection event.

For some game types where you can restart your computer and rejoin, everything works fine after rejoin (OpenGL).
